myia/api.py

```diff
@@ -115,6 +113,12 @@ def __call__(self, *args):
                 pass
         return self.compile(args)(*args)
 
+    def to_device(self, v, *, broaden=True, vm_t=None, orig_t=None):
+        """Move value to the function's accelerator hardware."""
+        backr = self.pip.steps['resources'].keywords['backend'].keywords
```

Comment:
I get it, but this looks pretty bad. Might be worth it to split resources into settings and resources, the former of which can be shared by all specializations in the MyiaFunction, but that's neither here nor there, I suppose.

Reply:
The problem is not sharing between specialization (since the backend will be shared anyway), but rather access to the values before any specialization has happened.  Therefore we don't have access to a built pipeline.